official/vision/beta/projects/yolo/dataloaders/yolo_input.py

```diff
@@ -0,0 +1,511 @@
+  """Function used to reduce COCO 91 to COCO 80, or to convert from the 2017 
+  foramt to the 2014 format"""
+  # Vector where index i coralates to the class at index[i].
+  x = [
+      1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
+      23, 24, 25, 27, 28, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43,
+      44, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62,
+      63, 64, 65, 67, 70, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 84, 85,
+      86, 87, 88, 89, 90
+  ]
+  no = tf.expand_dims(tf.convert_to_tensor(x), axis=0)
+
+  # Resahpe the classes to in order to build a class mask.
+  ce = tf.expand_dims(classif, axis=-1)
+
+  # One hot the classificiations to match the 80 class format.
+  ind = ce == tf.cast(no, ce.dtype)
+
+  # Select the max values.
+  co = tf.reshape(tf.math.argmax(tf.cast(ind, tf.float32), axis=-1), [-1])
+  ind = tf.where(tf.reduce_any(ind, axis=-1))
+
+  # Gather the valuable instances.
+  classif = tf.gather_nd(co, ind)
+  box = tf.gather_nd(box, ind)
+  areas = tf.gather_nd(areas, ind)
+  iscrowds = tf.gather_nd(iscrowds, ind)
+
+  # Restate the number of viable detections, ideally it should be the same.
+  num_detections = tf.shape(classif)[0]
+  return classif, box, areas, iscrowds, num_detections
+
+
+class Parser(parser.Parser):
+  """Parse the dataset in to the YOLO model format. """
+
+  def __init__(
+      self,
+      output_size,
+      masks,
+      anchors,
+      strides,
+      anchor_free_limits=None,
+      max_num_instances=200,
+      area_thresh=0.1,
+      aug_rand_hue=1.0,
+      aug_rand_saturation=1.0,
+      aug_rand_brightness=1.0,
+      letter_box=False,
+      random_pad=True,
+      random_flip=True,
+      jitter=0.0,
+      aug_scale_min=1.0,
+      aug_scale_max=1.0,
+      aug_rand_translate=0.0,
+      aug_rand_perspective=0.0,
+      aug_rand_angle=0.0,
+      anchor_t=4.0,
+      scale_xy=None,
+      best_match_only=False,
+      coco91to80=False,
+      darknet=False,
+      use_tie_breaker=True,
+      dtype='float32',
+      seed=None,
+  ):
+    """Initializes parameters for parsing annotations in the dataset.
+
+    Args:
+      output_size: `Tensor` or `List` for [height, width] of output image. The
+        output_size should be divided by the largest feature stride 2^max_level.
+      masks: `Dict[List[int]]` of values indicating the indexes in the 
+        list of anchor boxes to use an each prediction level between min_level 
+        and max_level. each level must have a list of indexes.  
+      anchors: `List[List[Union[int, float]]]` values for each anchor box.
+      strides: `Dict[int]` for how much the model scales down the images at the
+        largest level.
+      anchor_free_limits: `List` the box sizes that will be allowed at each FPN 
+        level as is done in the FCOS and YOLOX paper for anchor free box 
+        assignment. Anchor free will perform worse than Anchor based, but only 
```

Comment:
`List` the box sizes that will be allowed at level. The rest of description is model/training setup dependent, please remove.